app/assets/stylesheets/govuk_publishing_components/components/_layout-super-navigation-header.scss

```diff
@@ -23,103 +191,470 @@ $search-icon-size: 20px;
   }
 }
 
-.gem-c-layout-super-navigation-header__items {
+// Top level navigation and search.
+.gem-c-layout-super-navigation-header__navigation-items,
+.gem-c-layout-super-navigation-header__search-items {
+  background: govuk-colour("light-grey");
+  display: block;
   list-style: none;
-  margin: 0;
-  padding: 0 0 govuk-spacing(1) 0;
+  margin: 0 (0 - govuk-spacing(3));
+  padding: 0;
+
+  @include govuk-media-query($from: "tablet") {
+    margin: 0 (0 - govuk-spacing(6));
+  }
 
   @include govuk-media-query($from: "desktop") {
+    background: none;
+    float: left;
     padding: 0;
+    margin: 0;
   }
 }
 
-.gem-c-layout-super-navigation-header__item {
+.gem-c-layout-super-navigation-header__navigation-items {
+  .js-enabled & {
+    padding: 0 0 govuk-spacing(9) 0;
+
+    @include govuk-media-query($from: "desktop") {
+      padding: 0;
+    }
+  }
+}
+
+.gem-c-layout-super-navigation-header__navigation-item,
+.gem-c-layout-super-navigation-header__search-item {
   margin-right: 0;
-  padding: govuk-spacing(1) 0;
+  padding: 0;
 
   @include govuk-media-query($from: "desktop") {
+    border-bottom: none;
     float: left;
     padding: 0;
   }
 }
 
-.gem-c-layout-super-navigation-header__item-link {
-  display: block;
-  padding: govuk-spacing(3) 0;
-  position: relative;
+.gem-c-layout-super-navigation-header__navigation-item {
+  &:after {
```

Comment:
I think I tried, but because borders run from inside the margin but outside the padding it was difficult to get this right without overwriting some of the `govuk-grid-` styles.

I'll take another look with fresh eyes and see what can be done.

Reply:
I've taken a look at this, and I'd rather leave this as is for the time being.